src/build.cc

```diff
@@ -474,6 +486,154 @@ bool Plan::CleanNode(DependencyScan* scan, Node* node, string* err) {
   return true;
 }
 
+namespace {
+bool EdgeCom(const Edge* lhs, const Edge* rhs) {
+  // Note: > to sort in decreasing order.
+  return lhs->critical_time() > rhs->critical_time();
+}
+}  // namespace
+
+void Plan::ComputePriorityList(BuildLog* build_log) {
+
+  //testcase have no build_log
+  if (!build_log)
+    return;
+
+  METRIC_RECORD("ComputePriorityList");
+  set<Node*> dedup;
+  vector<Node*> deduped;
+  for (vector<Node*>::iterator it = targets_.begin(); it != targets_.end();
+       ++it) {
+    pair<set<Node*>::iterator, bool> insert_result = dedup.insert(*it);
+    if (!insert_result.second)
+      continue;
+    deduped.push_back(*it);
```

Comment:
Instead of calling `push_back` in a loop, it should be faster to call `targets_.assign(dedup.begin(), dedup.end()` afterwards.

Reply:
but it will break user input target order?

Reply:
why?

Reply:
set is order by Node* , but that maybe not the order user typed in command line.

Reply:
Ah right! Sorry that I've missed that.